tests/unit/test-extension-runners/test.extension-runners.firefox-desktop.js

```diff
@@ -0,0 +1,407 @@
+/* @flow */
+
+import {describe, it} from 'mocha';
+import {assert} from 'chai';
+import sinon from 'sinon';
+
+import {
+  FirefoxDesktopExtensionRunner,
+} from '../../../src/extension-runners/firefox-desktop';
+import {
+  getFakeFirefox,
+  getFakeRemoteFirefox,
+  makeSureItFails,
+  StubChildProcess,
+} from '../helpers';
+import {
+  onlyInstancesOf,
+  WebExtError,
+  RemoteTempInstallNotSupported,
+} from '../../../src/errors';
+
+// Fake result for client.installTemporaryAddon().then(installResult => ...)
+const tempInstallResult = {
+  addon: {id: 'some-addon@test-suite'},
+};
+// Fake missing addon id result for client.installTemporaryAddon
+const tempInstallResultMissingAddonId = {
+  addon: {id: null},
+};
+
+type PrepareParams = {
+  params?: Object,
+  deps?: Object,
+  fakeFirefoxApp?: Object,
+  fakeRemoteFirefox?: Object,
+  debuggerPort?: number,
+}
+
+function prepareExtensionRunnerParams({
+  params, fakeFirefoxApp, fakeRemoteFirefox, debuggerPort,
+}: PrepareParams = {}) {
+  const remoteFirefox = getFakeRemoteFirefox({
+    installTemporaryAddon: sinon.spy(
+      () => Promise.resolve(tempInstallResult)
+    ),
+    ...fakeRemoteFirefox,
+  });
+  const firefoxProcess = new StubChildProcess();
+
+  return {
+    remoteFirefox,
+    firefoxProcess,
+    params: {
+      extensions: [{
+        sourceDir: '/fake/sourceDir',
+        manifestData: {
+          name: 'fake-addon',
+          version: '0.1',
+          applications: {
+            gecko: {
+              id: 'test@fake.id',
+            },
+          },
+        },
+      }],
+      keepProfileChanges: false,
+      browserConsole: false,
+      startUrl: undefined,
+      firefoxBinary: 'firefox',
+      preInstall: false,
+      noReload: false,
+      firefoxApp: getFakeFirefox({
+        run: sinon.spy(() => {
+          return Promise.resolve({
+            debuggerPort,
+            firefox: firefoxProcess,
+          });
+        }),
+        ...fakeFirefoxApp,
```

Comment:
is this in the right order? I'm wondering if the `run` definition needs to come after it